### PR TITLE
test: add intg tests for tasks module

### PR DESCRIPTION
**Pull request by carolinaecalderon**
## Description
Add integration tests for all exported functions in `db/postgres_tasks.go`, and initialize the db in TestMain.

Add bun descriptions to the AllocationSession & TaskLog models, so that the ID key is auto-incremented with each addition.

## Test Plan

See attached test.

## Commentary (optional)
Truncated the start time in `RequireMockAllocation` to match what the db stores.
Remove `CheckTaskExists ` from `postgres_tasks.go`, as it's used nowhere.

Writing the tests first, then bun-ifying db/postgres_tasks.go in https://github.com/determined-ai/determined/pull/8764

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
DET-10124

### CODE REVIEW

File: master/internal/db/postgres_tasks_intg_test.go
Comment (salonig23) [inline master/internal/db/postgres_tasks_intg_test.go:655]: Can we test Tasklogs with a few api.Filters to see if we are getting the expected results?

File: master/internal/db/postgres_tasks_intg_test.go
Comment (salonig23) [inline master/internal/db/postgres_tasks_intg_test.go:607]: This test is failing for me, I ran it twice. The code seems correct to me but can you check on your end?

**Review by salonig23 — COMMENTED** [review]
A few comments but the rest looks great!

**Review by NicholasBlaskey — APPROVED** [review]
Looks good to me, nice work

I would wait for @salonig23 's approval too

File: master/internal/db/postgres_tasks_intg_test.go
Comment (carolinaecalderon) [reply master/internal/db/postgres_tasks_intg_test.go:607]: huh interesting, as it's passing on circleCI -- I think it's because your db isn't reset between runs, so an allocation that shouldn't exist, does. I'll rewrite the test to be unique on each run.

File: master/internal/db/postgres_tasks_intg_test.go
Comment (carolinaecalderon) [reply master/internal/db/postgres_tasks_intg_test.go:655]: added a test block

File: master/internal/db/postgres_tasks_intg_test.go
Comment (carolinaecalderon) [reply master/internal/db/postgres_tasks_intg_test.go:412]: No I guess not, I didn't write `TestNonExperimentTasksContextDirectory`, but definitely should've read it more closely.

File: master/internal/db/postgres_tasks_intg_test.go
Comment (carolinaecalderon) [reply master/internal/db/postgres_tasks_intg_test.go:36]: Why should we use SingleDB() instead of intializing the global variable after we start the db in TestMain?

Comment (NicholasBlaskey) [reply master/internal/db/postgres_tasks_intg_test.go:36]: I think the global variable is kinda duplicating what we already do in the code

I don't know if there is a misunderstanding of what I was suggesting but I don't mean to change any of the non-test code, I just mean in all functions where you deleted

```diff
-	require.NoError(t, etc.SetRootPath(RootFromDB))
-	db := MustResolveTestPostgres(t)
-	MustMigrateTestPostgres(t, db, MigrationsFromDB)
```

Just string replace replace ```db.``` with ```SingleDB().```

Do you think that would be too many changes for this PR?

**Review by salonig23 — APPROVED** [review]
LGTM, great work!

File: master/internal/db/postgres_tasks_intg_test.go
Comment (carolinaecalderon) [reply master/internal/db/postgres_tasks_intg_test.go]: Ah now I see -- it's just a 30-line change, so I'll include it in this PR.

**Pull request merged**
